### DISCUSSION TITLE
How the release code is composed?

### DISCUSSION THREAD

**Discussion by rrodrigueznt** [abraunegg/onedrive#2788] [General]
Hi! I've been trying to understand, and failing, how the version/release code is composed. I am mainly concern about understanding when a release is never than other by reading the version code. It is obvious that `2.5.0` > `2.4.45`, but why `onedrive v2.5.0-rc3-36-gf527eca` > `onedrive v2.5.0-rc3-2-g4b6b379`.

Particularly, what does represent `-36-` or `-2-` in the release codes above? And the seven digits after `-g`? I founded each commit has a seven digits identifier, but I'm not able to find the relationship among commitments and release codes.

Thanks!

**Comment by abraunegg** [comment]
@rrodrigueznt 
It is all to do with the commits done post a release.

Review this page carefully and corelate the numbers and hashes for each commit:

https://github.com/abraunegg/onedrive/compare/v2.5.0-rc3...onedrive-v2.5.0-release-candidate-3

**Reply by rrodrigueznt** [reply]
Got it! Thanks! Compiled a while ago:

```
(dmd-2.105.3)[rjr@e220i058h024l onedrive]$ ./onedrive --version
onedrive v2.5.0-rc3-38-gb211724
(dmd-2.105.3)[rjr@e220i058h024l onedrive]$
```
I had to click on [Releases](https://github.com/abraunegg/onedrive/releases), on the right column of the project's page. Then, click on the title [onedrive client version 2.5.0-rc3](https://github.com/abraunegg/onedrive/releases/tag/v2.5.0-rc3), and I got a page with a line including the number of commits since the release of the release.

![image](https://github.com/user-attachments/assets/c1ba25f4-d877-4126-a102-44b4a7ec0fcd)

Click on commits and go to the last in the page that is the most recent one:

![image](https://github.com/user-attachments/assets/03a0bb1c-926b-4fe0-bd73-a357a7dca42e)

Thanks!
